src/gui/qgsadvanceddigitizingtools.cpp

```diff
@@ -0,0 +1,299 @@
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <QMouseEvent>
+#include <QEnterEvent>
+#include <QLocale>
+
+#include "qgsadvanceddigitizingtools.h"
+#include "qgsapplication.h"
+#include "qgsdoublespinbox.h"
+#include "qgsmapcanvas.h"
+
+QgsAdvancedDigitizingTool::QgsAdvancedDigitizingTool( QgsMapCanvas *canvas, QgsAdvancedDigitizingDockWidget *cadDockWidget )
+  : QObject( canvas ? canvas->viewport() : nullptr )
+  , mMapCanvas( canvas )
+  , mCadDockWidget( cadDockWidget )
+{
+}
+
+QgsAdvancedDigitizingCirclesIntersectionTool::QgsAdvancedDigitizingCirclesIntersectionTool( QgsMapCanvas *canvas, QgsAdvancedDigitizingDockWidget *cadDockWidget )
+  : QgsAdvancedDigitizingTool( canvas, cadDockWidget )
+{
+}
+
+QgsAdvancedDigitizingCirclesIntersectionTool::~QgsAdvancedDigitizingCirclesIntersectionTool()
+{
+  if ( mToolWidget )
+  {
+    mToolWidget->deleteLater();
+  }
+}
+
+QWidget *QgsAdvancedDigitizingCirclesIntersectionTool::createWidget()
+{
+  QWidget *toolWidget = new QWidget();
+
+  QGridLayout *layout = new QGridLayout( toolWidget );
+  layout->setContentsMargins( 0, 0, 0, 0 );
+  toolWidget->setLayout( layout );
+
+  QLabel *label = new QLabel( QStringLiteral( "Circle #1" ), toolWidget );
+  layout->addWidget( label, 0, 0, 1, 3 );
+
+  mCircle1Digitize = new QToolButton( toolWidget );
+  mCircle1Digitize->setCheckable( true );
+  mCircle1Digitize->setChecked( false );
+  mCircle1Digitize->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionMapIdentification.svg" ) ) );
+  connect( mCircle1Digitize, &QAbstractButton::toggled, this, [ = ]( bool checked )
+  {
+    if ( checked )
+    {
+      mCircle2Digitize->setChecked( false );
+    }
+  } );
+  layout->addWidget( mCircle1Digitize, 1, 2, 2, 1 );
+
+  label = new QLabel( QStringLiteral( "X" ), toolWidget );
+  layout->addWidget( label, 1, 0 );
+
+  mCircle1X = new QgsDoubleSpinBox( toolWidget );
+  mCircle1X->setMinimum( std::numeric_limits<double>::min() );
+  mCircle1X->setMaximum( std::numeric_limits<double>::max() );
+  connect( mCircle1X, &QgsDoubleSpinBox::textEdited, this, [ = ]() { mCircle1Digitize->setChecked( false ); } );
+  layout->addWidget( mCircle1X, 1, 1 );
+
+  label = new QLabel( QStringLiteral( "Y" ), toolWidget );
+  layout->addWidget( label, 2, 0 );
+
+  mCircle1Y = new QgsDoubleSpinBox( toolWidget );
+  mCircle1Y->setMinimum( std::numeric_limits<double>::min() );
+  mCircle1Y->setMaximum( std::numeric_limits<double>::max() );
+  connect( mCircle1Y, &QgsDoubleSpinBox::textEdited, this, [ = ]() { mCircle1Digitize->setChecked( false ); } );
+  layout->addWidget( mCircle1Y, 2, 1 );
+
+  label = new QLabel( QStringLiteral( "Distance" ), toolWidget );
```

Comment:
`Distance` is too long string (also not translated) causing the ugly widget to grow unnecessarily, let's change that to `R` for radius or `d` for distance, which is what is used in the advanced digitizing panel.